### DISCUSSION TITLE
smtp host

### DISCUSSION THREAD

**Discussion by mrsauron69** [pH7Software/pH7-Social-Dating-CMS#984] [Troubleshooting]
I see some problems with sending email . but don't see any solution for this yet.

Oops! Our email server encountered an internal error. The email could not be sent. Please try again later

How can you put the smtp data in the script instead of the phpmailer?

can not change the php.ini because im using a shared hosting.

thanks
Sauron

**Comment by MAD-King** [comment]
Having the same issue. Need to use SMTP but no option for this.

**Comment by GeoffCee** [comment]
Seems like there is no SMPT config in php7builder. Would be good to have this as a config option to use SMPT credentials. It is not clear how to set up any server on aws or digital ocean to use mailgun as the email sender for php7builder.

**Comment by fshahzad** [comment]
As per this file code, the SMTP settings would be stored from database, if there isn't any provide while calling the Mail\send() function

https://github.com/pH7Software/pH7-Social-Dating-CMS/blob/18.x/_protected/framework/Mail/Mail.class.php

**Reply by GeoffCee** [reply]
So you put the SMPT details in the database? then the script will use those settings?

**Comment by fshahzad** [comment]
Login to Admin panel and then Goto "Settings" => "General Settings" then click and open the "Emails" tab

**Reply by GeoffCee** [reply]
These are the email addresses to use, not the SMPT details. Are you are saying if these are blank, it will use this SMPT setting in the db?

**Reply by fshahzad** [reply]
Well in this case, this codebase clearly lacks the feature of setting the SMTP via any form either by database or by modifiying any php code file.

**Comment by fshahzad** [comment]
- I have modified the codebase now and introduced the settings of SMTP in the 'General settings' -> 'Email' tab
- so now Mail class checks for database 'mailType' and construct a MailTransport SMTP via databaes settings
<img width="2400" height="2542" alt="Screenshot 2026-01-27 at 12-46-41 General Settings - Local Dating Site" src="https://github.com/user-attachments/assets/93148678-0f16-4874-8973-2d4ed0cbea50" />

**Reply by fshahzad** [reply]
Branch 18.x

**Comment by fshahzad** [comment]
https://github.com/pH7Software/pH7-Social-Dating-CMS/pull/1188

**Comment by pH-7** [comment]
Hi! pH7Builder v19 supports SMTP through `PH7_MAILER_DSN`, set in your site's PHP environment. The admin Email tab sets email addresses, not SMTP credentials.

On shared hosting, ask your host to configure this variable. Here is the [SMTP setup guide](https://github.com/pH7Software/pH7-Social-Dating-CMS/blob/18.x/docs/QUICK_START.md#8-configure-mail-delivery). Please keep passwords private.

Best, [Pierre-Henry](https://github.com/pH-7)
